lib/LaTeXML/Post/MathML.pm

```diff
@@ -670,89 +645,56 @@ my %plane1hackable = (    # CONSTANT
-  my $text     = (ref $item  ? $item->textContent              : $item);
-  my $variant  = ($font      ? unicode_mathvariant($font)      : '');
-  my $stretchy = ($iselement ? $item->getAttribute('stretchy') : $attr{stretchy});
-  $stretchy = undef if ($tag ne 'm:mo');                    # Only allowed on m:mo!
-  $size     = undef if ($stretchy || 'false') eq 'true';    # Ignore size, if we're stretching.
-
-  my $stretchyhack = undef;
-
-  if ($text =~ /^[\x{2061}\x{2062}\x{2063}]*$/) {           # invisible get no size or stretchiness
-    $stretchy = $size = undef; }
-  if ($size) {
-    if ($size eq ($LaTeXML::MathML::SIZE || 'text')) {      # If default size, no need to mention.
-      $size = undef; }
-    # If requested relative size, and in script or scriptscript, we'll need to adjust the size
-    elsif (($size =~ /%$/) && ($LaTeXML::MathML::STYLE and $LaTeXML::MathML::STYLE =~ /script/)) {
-      my $req = $size;                               $req =~ s/%$//;
-      my $ex  = $stylesize{$LaTeXML::MathML::STYLE}; $ex  =~ s/%$//;
-      $size = int(100 * $req / $ex) . '%'; }
-    # Note that symmetric is only allowed when stretchy, which looks crappy for specific sizes
-    # so we'll pretend that delimiters are still stretchy, but restrict size by minsize & maxsize
-    # (Thanks Peter Krautzberger)
-    # Really we should check the Operator Dictionary to see if it's expected to be symmetric
-    if ($size) {
-      if ($role && $symmetric_roles{$role}) {
-        $stretchyhack = 1;
-        $stretchy     = undef; }
-      elsif ($tag eq 'm:mo') {
-        $stretchy = 'false'; } } }    # Conversely, if size specifically set, don't stretch it!
-  elsif ($normally_stretchy{$text}) {    # Else, if this would normally be stretchy
-    if ($stretchy && ($stretchy eq 'true')) {
-      $stretchy = undef; }               # Don't need to say this explicitly
-    else {
-      $stretchy = 'false'; } }           # or need to explicitly disable it.
-  elsif ($stretchy && ($stretchy eq 'false')) {
-    $stretchy = undef; }                 # Otherwise, doesn't need to be said at all.
+  my $class = join(' ', grep { $_; } $attr{class}, ($iselement && $item->getAttribute('class')));
+  my $cssstyle = join('; ', grep { $_; } $attr{ccsstyle}, ($iselement && $item->getAttribute('cssstyle')));
+  my $variant = ($font ? unicode_mathvariant($font) : '');
+  my $istoken = $tag =~ /^m:(?:mi|mo|mn)$/;                  # mrow? well, no....
+  my $href    = $istoken && ($attr{href}  || ($iselement && $item->getAttribute('href')));
+  my $title   = $istoken && ($attr{title} || ($iselement && $item->getAttribute('title')));
+
+  # Implied attributes, relevant for mo operators
+  my $stretchy = ((defined $attr{stretchy} ? $attr{stretchy} : ($iselement && $item->getAttribute('stretchy')))
+      || 'false') eq 'true';
+  my $isfence   = $role && ($role =~ /^(OPEN|CLOSE|MIDDLE)$/);
+  my $ispunct   = $role && ($role eq 'PUNCT');
```

Comment:
Unsure if worth mentioning but `PERIOD` also acts as punctuation in the `endPunct` grammar rule. Would it be helpful to recognize here?

Reply:
Probably better to call it `$issep`, which is how it's used (upcoming commit)